python/casmsmartphase/MNVMerge.py

```diff
@@ -0,0 +1,313 @@
+        and (UTC) date appended to the key
+        """
+        head_key = base_vcf_process_key
+        index = get_last_vcf_process_index(existing_head.lines, head_key)
+        # Test for key without date - gives all lines
+        if index is not None and index > 0:
+            now = datetime.datetime.utcnow()
+            date_str = now.strftime("%Y%m%d")
+            head_key += "_{}".format(date_str)
+            # Test for key with current date to get precise index
+            index = get_last_vcf_process_index(existing_head.lines, head_key)
+            if index is None:
+                index = 0
+            head_key = head_key + "." + str(index + 1)
+        new_process_line = vcfpy.HeaderLine(
+            key=head_key,
+            value=base_vcf_process_log.format(
+                self.vcfinname, self.run_script, self.arg_str
+            ),
+        )
+        return new_process_line
+
+    def generate_new_increment_header(self, existing_line, n):
+        """
+        Taking a header line, and an int n, generates a copy of that header
+        line with the key and description updates to include said
+        incremental int
+        """
+        new_line = existing_line.copy()
+        new_line.mapping["ID"] = new_line.mapping["ID"] + f"_{n}"
+        new_line.mapping["Description"] = (
+            new_line.mapping["Description"] + f" (MNV allele {n} in series)"
+        )
+        return new_line
+
+    def parse_header_add_merge_and_process(self, writer_header):
+        """
+        Parse VCF header from input. Add a process line and MNV
+        format lines
+        """
+        # Add a headerline to say this was refiltered with this tool
+        process_head_line = self.get_process_header_line(writer_header)
+        writer_header.add_line(process_head_line)
+        info_lines = writer_header.get_lines("INFO")
+        format_lines = writer_header.get_lines("FORMAT")
+        # Add a X1..X2..Xn info and format header lines for
+        lines_to_add = []
+        for inf_line in info_lines:
+            for i in range(1, self.longest_MNV + 1):
+                lines_to_add.append(self.generate_new_increment_header(inf_line, i))
+        for form_line in format_lines:
+            for i in range(1, self.longest_MNV + 1):
+                lines_to_add.append(self.generate_new_increment_header(form_line, i))
+
+        for new_head_line in lines_to_add:
+            writer_header.add_line(new_head_line)
+
+        return writer_header
+
+    def merge_snv_to_mnv(self, snv_list):
+        """
+        Merge snvs from list into a single variant and output to VCF
+        """
+        do_qual = 0
+        do_filter = 0
+        qual = 0
+        filter = []
+        if snv_list[0].QUAL:
+            do_qual = 1
+            print(
+                "Found a QUAL value, output will be a mean of all QUAL.",
+                file=sys.stderr,
+                end="\n",
+            )
```

Comment:
Why not python logging, it's trivial to use compared to `log4*`?  Also, don't define params for defaults (`end="\n"`)